File: decompiled/sources/com/mdvr/BlackBox/BlackBoxFrame.java
package com.mdvr.BlackBox;

import android.util.Log;

public class BlackBoxFrame {
    private static final String TAG = "BlackBoxFrame";
    public DateTime datetime;
    public GPSInfo gpsInfo;
    long ullFrameStamp;

    public void print() {
        String format = String.format("%04d-%02d-%02d %02d:%02d:%02d", new Object[]{Integer.valueOf(this.datetime.cYear + 2000), Byte.valueOf(this.datetime.cMonth), Byte.valueOf(this.datetime.cDay), Byte.valueOf(this.datetime.cHour), Byte.valueOf(this.datetime.cMinute), Byte.valueOf(this.datetime.cSecond)});
        Log.v(TAG, "[" + format + "]" + this.gpsInfo.usSpeed);
    }
}

File: decompiled/sources/com/mdvr/BlackBox/BlackBoxFrame.kt
package com.mdvr.BlackBox

import android.util.Log

class BlackBoxFrame {
    @JvmField
    var datetime: DateTime? = null

    @JvmField
    var gpsInfo: GPSInfo? = null

    @JvmField
    var ullFrameStamp: Long = 0

    fun print() {
        val dt = datetime ?: return
        val gps = gpsInfo ?: return
        val format = String.format(
            "%04d-%02d-%02d %02d:%02d:%02d",
            dt.cYear.toInt() + 2000,
            dt.cMonth.toInt(),
            dt.cDay.toInt(),
            dt.cHour.toInt(),
            dt.cMinute.toInt(),
            dt.cSecond.toInt()
        )
        Log.v(TAG, "[$format]${gps.usSpeed}")
    }

    companion object {
        private const val TAG = "BlackBoxFrame"
    }
}
